lib/blockchain/chain.js

```diff
@@ -378,7 +378,7 @@ class Chain extends AsyncEmitter {
         if (!block.witnessRoot.equals(witnessRoot)) {
           throw new VerifyError(block,
             'invalid',
-            'bad-txnmrklroot',
+            'bad-witnessroot',
```

Comment:
am i correctly understanding these strings are just identifiers for the message lookup? seems to me like they should probably be globally constantized which I figure can prevent certain kinds of mistakes and may have other slight benefits like performance wise.

Reply:
s/prevent/help catch early/

Reply:
also, is there a good reason for abbreviating? seems harder to read/write/remember to me.

Reply:
No these are the actual error messages themselves. You can see the `VerifyError` class defined here:

https://github.com/handshake-org/hsd/blob/505b6f76834ef93ee883b8052780936c92f199fa/lib/protocol/errors.js#L42

So these strings go directly to the debug log but are also transmitted in `RejectPacket` back to the peer that sent the bad message:

https://github.com/handshake-org/hsd/blob/505b6f76834ef93ee883b8052780936c92f199fa/lib/net/packets.js#L878


(Luckily) I couldn't find any bad block messages in my own debug log but the same mechanism is used for transactions as well as blocks:

```
[D:2021-08-24T23:53:35Z] (peer) Rejecting CLAIM 1db79beb3b0144334e1fe7b2f390770ab61d2c7dc25077ae3c8b0748954b24ed (103.121.199.186:55007): code=alreadyknown reason=name-already-in-mempool.
[D:2021-08-24T23:54:01Z] (peer) Rejecting TX 885df0431fa791a6acd0cfacf4b2fd690f762ffde510df8a680ec8e743f1e40e (44.240.154.53:54918): code=duplicate reason=bad-txns-inputs-spent.
[D:2021-08-25T02:53:38Z] (peer) Rejecting TX 9e08ba19b3491dde5f83ba8ff23dc48f696458a31cfc792a64fb387ee26797df (44.240.154.53:54918): code=duplicate reason=bad-txns-inputs-spent.
[D:2021-08-25T02:55:24Z] (peer) Rejecting TX 02ea9e5c4535361775d359eedbf78bbf3df197da8e6f29b0ca7367d912b6f4f8 (44.240.154.53:54918): code=duplicate reason=bad-txns-inputs-spent.
[D:2021-08-25T05:01:05Z] (peer) Rejecting TX 0717e68d92102037b10a51de98cefc8fd3d9bcc10951a9ceec3ab3afb2919e8b (129.213.167.123:44886): code=alreadyknown reason=txn-already-known.
[D:2021-08-25T05:02:40Z] (peer) Rejecting TX 0c0aab6743b0d231b12c47a4411223c5209d53e8d447433821fc039cf97f2d3c (44.240.154.53:54918): code=duplicate reason=bad-txns-inputs-spent.
[W:2021-08-25T05:49:31Z] (net) Received reject (47.108.193.203:35420): msg=TX code=invalid reason=invalid-covenant hash=6a971dd53bc007c1bb3f698797bf46e17e4f6c478a2a414fe14792f6789e9383.
[W:2021-08-25T05:49:31Z] (net) Received reject (47.108.193.203:35420): msg=TX code=invalid reason=invalid-covenant hash=f40ebd32ce50558b7becad34f8dc68f4d8c79522e5a5dd5cfe73c0feb61c8c77.
[D:2021-08-25T05:51:59Z] (peer) Rejecting TX fcb59773709f6b1c6e3a2e83f171ef82886a811f233191b61df9b27cff350532 (44.240.154.53:54918): code=duplicate reason=bad-txns-inputs-spent.
[D:2021-08-25T05:51:59Z] (peer) Rejecting TX fd80f6343a4c35b0894b823d9ca6450fa8cfa1b6d67b5fb41a7aaea9b1026b97 (44.240.154.53:54918): code=duplicate reason=bad-txns-inputs-spent.
[D:2021-08-25T06:51:06Z] (peer) Rejecting TX a980c801360abeef19e952fb2b2af4438127b34ef386a2c5f78e376016c3e333 (89.217.249.119:49953): code=invalid reason=invalid-covenant.
[D:2021-08-25T06:51:06Z] (peer) Rejecting TX c4ab2fa654c884809c56a605367fda45d991ff1bbce953006a98261e8dfec3f7 (89.217.249.119:49953): code=invalid reason=invalid-covenant.
```